rfc_samplers.md

```diff
@@ -0,0 +1,107 @@
+**DISCLAIMER** This RFC is currently lacking (at least):
+
+- a "time-based" equivalent of dilation
+- a "time-based" equivalent of random uniform sampling.
+
+Main design principles
+----------------------
+
+- The existing feature space of different libraries (torchvision, torchmm,
+  internal stuff) is supported. In particular "Random", "Uniform" and
+  "Periodic" samplers are supported, with hopefully more descriptive names.
+- Sampler is agnostic to decoding options: the decoder object is passed to the
+  sampler by the user.
+- Explicit **non goal**: to support arbitrary strategy that we haven't
+  observed or heard usage of. For example, in all existing libraries the clip
+  content is never random and just determined by 2 parameters:
+  `num_frames_per_clip` and `step_between_frames` (dilation). We allow for
+  future extensibility, but we're not explicitly enabling additional
+  clip-content strategies.
+- The term "clip" is being used to denote a sequence of frames in presentation
+  order. The frames aren't necessarily consecutive (when
+  `step_between_frames > 1`). We should add this term to the glossary. I don't
+  feel too strongly about using the name "clip", but it's used in all existing
+  libraries, and I don't know of an alternative that would be universally
+  understood.
+
+
+Option 1
+--------
+
+- One function per sampling strategy
+- Note: this is not 100% stateless, the decoder object is seeked so there are
+  side effects.
+
+```py
+from torchcodec.decoders import SimpleVideoDecoder
+from torchcodec import samplers
+
+video = "cool_vid"
+decoder = SimpleVideoDecoder(video)
+
+clips = samplers.get_uniformly_random_clips(
+    decoder,
+    num_clips=12,
+    # clip content params:
+    num_frames_per_clip=4,
+    step_between_frames=2,  # often called "dilation"
+    # sampler-specific params:
+    prioritize_keyframes=False,  # to implement "fast" sampling.
```

Comment:
I can see this as a parameter or a different top-level function. As to name, I think we would want something more about the perceivable qualities to users: accuracy versus speed. We could implement that by prioritizing keyframes or not.